src/components/implementation/no_interface/scheddev/sl.h

```diff
@@ -73,11 +73,25 @@ sl_thd_setprio(struct sl_thd *t, tcap_prio_t p)
 
 static inline struct sl_thd *
 sl_thd_lkup(thdid_t tid)
-{ return sl_mod_thd_get(sl_thd_lookup_backend(tid)); }
+{
+	assert(tid != 0);
+	if (unlikely(tid > MAX_NUM_THREADS)) return NULL;
+	return sl_mod_thd_get(sl_thd_lookup_backend(tid));
+}
+
+static inline thdid_t
+sl_thdid(void)
+{
+	thdid_t tid = cos_thdid();
+	assert(tid != 0);
+	assert(tid < MAX_NUM_THREADS);
+	return tid;
+}
+
 
 static inline struct sl_thd *
 sl_thd_curr(void)
-{ return sl_thd_lkup(cos_thdid()); }
+{ return sl_thd_lkup(sl_thdid()); }
```

Comment:
This is the case where safety checks would've really helped me with the cFE, so I'm reluctant to change this.